psychopy/app/jobs.py

```diff
@@ -588,9 +597,13 @@ def onTerminate(self, exitCode):
         # unbind the idle loop used to poll the subprocess
         self.parent.Bind(wx.EVT_IDLE, None)
 
+        self._readPipes()  # read remaining data
+
         # stop the pipe reader threads now
         self._stdoutReader.stop()
-        self._stderrReader.stop()
+        self._stdoutReader.stop()
+        self._stderrReader.join(timeout=1)
+        self._stderrReader.join(timeout=1)
```

Comment:
I don't get it. You call `stdoutReader.stop()` twice and then `stderrReader.join()` twice. Shouldn't each pipe stop once, and then each pipe joins once?